### DISCUSSION TITLE
So, node uses 100 cpu after ddos attack

### DISCUSSION THREAD

**Discussion by samik3k** [nuxt/nuxt#18403] [Questions (Nuxt 2)]
Hello all. Guys i need help.

"nuxt": "^2.13.3"
"vue": "^2.6.11"
nodejs 14
mode: 'universal',

I have 6 axios requests in nuxtServerInit.
So, form 14 january my application under DDOS attack.
After ddos attack node eats 100 cpu for 10-24 minutes.
Like some queue of request or i dont know
Anyone know where to look why its happens?

**Comment by samik3k** [comment]
Ok i found out about node profiler, how to run it with nuxtjs?

**Reply by samik3k** [reply]
`node ./node_modules/.bin/nuxt --prof`
Ok its started but no log

**Comment by Baroshem** [comment]
To mitigate this kind of issues, maybe you could implement a rate limiting middleware in your nuxt or server app or a fail2ban on an infrastructure layer?

https://www.npmjs.com/package/express-rate-limit

https://www.fail2ban.org/wiki/index.php/Main_Page

**Reply by samik3k** [reply]
Our devops said he dosnt want to ban IPs... Now im looking for nuxtjs 2 + prof flag to get nodejs profiler.
Maybe i can lower CPU usage after DDOS attack.

**Reply by Baroshem** [reply]
What about this then? https://www.npmjs.com/package/toobusy-js

**Reply by samik3k** [reply]
> What about this then? https://www.npmjs.com/package/toobusy-js

I need tool to find out what to fix to lower this 10-24 minutes of 100% cpu after ddos attack.
Not to inform that we are busy or whatever

**Reply by Baroshem** [reply]
Wouldn't it be better to react before being affected by ddos attack (by implementing rate limiting or request size limiting) than trying to fix it afterwards?

**Reply by samik3k** [reply]
> Wouldn't it be better to react before being affected by ddos attack (by implementing rate limiting or request size limiting) than trying to fix it afterwards?

I think that persons who doing ddos attack someday will ask for ransom or will increase attack power.
So our guys anyway will be need to spend more money for antiddos...
But memory leak or cpu usage need to be fixed.

Btw, i will read about limitation in nuxtjs2, maybe i will find something that helps

**Comment by ndragun92** [comment]
What about nuxt security module?
https://github.com/Baroshem/nuxt-security

For example, Vercel and certain host providers offer a big better DDoS protection than the basic one.

**Reply by Baroshem** [reply]
My module does not work for Nuxt 2 as it uses nitro event handlers for security middlewares. That is why I was not recommending it in the first place 😉

**Reply by ndragun92** [reply]
Oh now i realize it is ur module :D
